src/Microsoft.Extensions.ProjectModel.Sources/Internal/DotNetCoreSdkResolver.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Extensions.ProjectModel.Internal
+{
+    internal class DotNetCoreSdkResolver
+    {
+        private readonly string _installationDir;
+
+        /// <summary>
+        /// Instantiates a resolver that locates the SDK
+        /// </summary>
+        /// <param name="installationDir">The directory containing dotnet muxer, aka DOTNET_HOME</param>
+        public DotNetCoreSdkResolver(string installationDir)
+        {
+            _installationDir = installationDir;
+        }
+
+        /// <summary>
+        /// Find the latest SDK installation (uses SemVer 1.0 to determine what is "latest")
+        /// </summary>
+        /// <returns>Path to SDK root directory</returns>
+        public DotNetCoreSdk ResolveLatest()
+        {
+            var latest = FindInstalled()
+                .Select(d => new { path = d, version = SemanticVersion.Parse(Path.GetFileName(d)) })
+                .OrderByDescending(sdk => sdk.version)
+                .FirstOrDefault();
+
+            if (latest == null)
+            {
+                throw CreateSdkNotInstalledException();
+            }
+
+            return new DotNetCoreSdk
+            {
+                BasePath = latest.path,
+                Version = latest.version.ToFullString()
```

Comment:
`NuGet.Versioning.SematicVersion` is an implementation detail of how we figure out which version is "latest". In theory, SDK versions may not be valid `SemanticVersion`s.